src/components/common/Header.tsx

```diff
@@ -0,0 +1,107 @@
+import { IcCarWhite, IcFlagKorea, IcFlightWhite, IcHotelWhite, IcLikePage, IcMainLogo, IcMy } from '@/assets/svg';
+import FilterButton from '@/components/common/FilterButton.tsx';
+import { PATH } from '@/router/path.ts';
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+const Header = () => {
+	const navigate = useNavigate();
+	const location = useLocation();
+
+	const pathsWithFilter = ['/', '/wish'];
+	const isFilter = pathsWithFilter.includes(location.pathname);
+	const [selectedFilter, setSelectedFilter] = useState<string | null>(null);
+	const handleFilterClick = (filter: string) => {
+		setSelectedFilter(filter === selectedFilter ? null : filter);
+	};
+
+	return (
+		<HeaderContainer>
+			<HeaderTop>
+				<MainLogo onClick={() => navigate(PATH.HOME)} />
+				<Menu>
+					<NationWrap>
+						<NationItem>KO</NationItem>
+						<IcFlagKorea />
+						<NationItem>KRW(₩)</NationItem>
+					</NationWrap>
+					<IcLikePage onClick={() => navigate(PATH.WISH)} />
+					<IcMy />
+				</Menu>
+			</HeaderTop>
+			{isFilter && (
+				<HeaderFilter>
```

Comment:
p3: pr남겨주신 내용을 보고 더 좋은 방법은 없을까? 고민해봤는데 `HeaderFilter`에 `visible prop` 추가해주고, 스타일링 해주는 부분에서 `const HeaderFilter = styled.div<{ visible: boolean }>`과 같이 `visible`을 `prop`을 전달받아 `display: ${({ visible }) => (visible ? 'flex' : 'none')};`과 같이 동적으로 스타일링 해주는 방법도 있을거 같아요. 민정님 의견은 어떠신가요?
```suggestion
				<HeaderFilter visible={isFilter}>

```

Reply:
넵 의견 감사합니다.
해당 방법이 좀 더 깔끔할 것 같아서 수정해서 푸시했습니다!